subprojects/core/src/main/java/org/gradle/internal/service/scopes/ToPlannedNodeConverterRegistry.java

```diff
@@ -19,25 +19,26 @@
 import com.google.common.collect.ImmutableList;
 import org.gradle.execution.plan.Node;
 import org.gradle.execution.plan.ToPlannedNodeConverter;
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType;
+import org.gradle.internal.taskgraph.NodeIdentity;
 
 import javax.annotation.Nullable;
-import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 /**
  * A Gradle user home level registry of {@link ToPlannedNodeConverter} instances.
  * <p>
- * All the available converters are expected to support disjoined set of {@link Node node types}.
+ * All the available converters are expected to support disjoint set of {@link Node node types}.
  */
 public class ToPlannedNodeConverterRegistry {
```

Comment:
You could add a unit test for the concurrency behaviour of the class, though I don't think it is necessary.